build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,180 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="compile/Microsoft.DotNet.Cli.Archive.targets" />
+
+  <UsingTask TaskName="FixModeFlags" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="ChangeEntryPointLibraryName" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="RemoveAssetFromDepsPackages" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CleanPublishOutput" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="SharedFrameworkNameVersionPath" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CrossgenDirectory" AssemblyFile="$(CLIBuildDll)" />    
+
+  <Target Name="Compile" DependsOnTargets="BuildDotnetCliBuildFramework;
+                                           Init;
+                                           SetupStage;
+                                           CompileStage;
+                                           BuildProjectsForNuGetPackages;
+                                           GenerateNuGetPackagesArchive">
+  </Target>
+
+  <Target Name="SetupStage">
+    <ItemGroup>
+      <CompileStageInputs Include="$(RepoRoot)/src/**/*.cs" Exclude="$(RepoRoot)/src/**/obj/**/*.cs" />
+      <CompileStageInputs Include="$(RepoRoot)/src/**/project.json;$(RepoRoot)/src/**/project.lock.json" />
+
+      <Stage Include="Stage1">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage1Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage1SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(Stage0Path)/dotnet.exe</DotnetExe>
+      </Stage>
+      <Stage Include="Stage2">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage2Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage2SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(DotnetStage1)</DotnetExe>
+      </Stage>          
+    </ItemGroup>        
+  </Target>    
+
+  <Target Name="CompileStage"
+          Inputs="%(Stage.Identity);%(Stage.Inputs)"
+          Outputs="%(StageDirectory)/sdk/$(NugetVersion)/dotnet.dll">
+      <PropertyGroup>
+        <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
```

Comment:
If you move this property up above "SetupStage", you can use it inside SetupStage, and don't need to duplicated in so many spots.